packages/next/src/server/app-render/action-handler.ts

```diff
@@ -306,7 +307,8 @@ async function createRedirectRenderResult(
     const forwardedHeaders = getForwardedHeaders(req, res)
     forwardedHeaders.set(RSC_HEADER, '1')
 
-    const proto = workStore.incrementalCache?.requestProtocol || 'https'
+    const proto =
+      getRequestMeta(req, 'initProtocol')?.replace(/:$/, '') || 'https'
```

Comment:
maybe overly defensive but in case the input somehow ends up with two colons
```suggestion
      getRequestMeta(req, 'initProtocol')?.replace(/:+$/, '') || 'https'
```